packages/flutter/lib/src/widgets/nested_scroll_view.dart

```diff
@@ -291,6 +293,9 @@ class NestedScrollView extends StatefulWidget {
   /// or separate and desirous of unique behaviors.
   final ScrollBehavior? scrollBehavior;
 
+  /// Whether or not the [NestedScrollView] should use [NestedScrollController].
```

Comment:
I am hesitant about this, it seems to add a lot more complexity to this already complex widget.

Reply:
I have a question. I have seen this in `box.dart`
```dart 
void setupParentData(covariant RenderObject child) {
    if (child.parentData is! BoxParentData) {
      child.parentData = BoxParentData();
    }
  } 
```
Why is it allowed but not
``` dart
bool get _useNestedScrollController => widget.controller is NestedScrollController
```

Reply:
This is from our style guide: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#avoid-interleaving-multiple-concepts-together

Some recent chatter about it too: https://discord.com/channels/608014603317936148/608018585025118217/979083546989838366